### PR TITLE
Rebecca steel hall

### PR DESCRIPTION
**Pull request by Rebecca-Coding21**
# Description

I added the example "Steel_Hall" in the folder "examples". It's a programm to generate a steel hall with several frames and bracings according to user-input. The example contains five loadcases and two load combinations. After calculation the results are exported to csv-files.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual Testing with different user-inputs

**Test Configuration**:
* RFEM / RSTAB version: 6.02.0053
* .Net SDK: .NET 6.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: examples/Steel_Hall/Program.cs [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+                mesh_statistics_type mesh_Statistics = model.get_mesh_statistics();
+                Console.WriteLine("Number of mesh nodes: " + mesh_Statistics.node_elements);
+                Console.WriteLine("Number of 1D elements: " + mesh_Statistics.member_1D_finite_elements);
+                Console.WriteLine("Number of surface element: " + mesh_Statistics.surface_2D_finite_elements);
+                Console.WriteLine("Number of volume elements: " + mesh_Statistics.solid_3D_finite_elements);
+                #endregion
+#endif
+                calculation_message[] calculationMessages = model.calculate_all(true);
+                if (calculationMessages.Length != 0)
+                {
```
Comment (jarabroz) [inline examples/Steel_Hall/Program.cs:1208]: You can add here printing of calculation message if something is unsucessful during calculation

File: examples/Steel_Hall/Program.cs [hunk 2]
```diff
@@ -0,0 +1,1291 @@
...
+                    Console.WriteLine("Model has LC5 results.");
+                }
+                else
+                {
+                    Console.WriteLine("Model has no LC5 results.");
+                }
+                #endregion
+
+                #region export results
+                model.export_result_tables_with_detailed_members_results_to_csv("C:\\Users\\GoebelR\\Documents\\Webservices\\testmodels\\CSV");
```
Comment (jarabroz) [inline examples/Steel_Hall/Program.cs:1270]: Don't add absolute path use e.g. 
```suggestion
                string currentDir = Directory.GetCurrentDirectory();
                model.export_result_tables_with_detailed_members_results_to_csv(currentDir + @"\CSV\");
```

File: examples/Continuous_Beam/Program.cs [hunk 3]
```diff
@@ -0,0 +1,642 @@
...
+                    else
+                    {
+                        Console.WriteLine("Total \t \t \t \t quantity: {4}\t length: {5}\t unit surface area: {6}\t volume: {7}\t unit mass: {8}\t member mass: {9}\t total length: {10}\t total surface area: {11}\t total volume:{12}\t total mass:{13}",
+                                            item.description, item.row.material_name, item.row.section_name, item.row.members_no, item.row.quantity, item.row.length, item.row.unit_surface_area, item.row.volume, item.row.unit_mass, item.row.member_mass, item.row.total_length, item.row.total_surface_area, item.row.total_volume, item.row.total_mass);
+                    }
+                }
+                #endregion
+
+                //save the model before closing
+                model.save("C:\\Users\\GoebelR\\Documents\\Webservices\\testmodels\\testmodel");
```
Comment (jarabroz) [inline examples/Continuous_Beam/Program.cs:632]: Don't use absolute path

**Review by jarabroz — CHANGES_REQUESTED** [review]

File: examples/Steel_Hall/Program.cs [hunk 4]
```diff
@@ -0,0 +1,1291 @@
...
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                Console.WriteLine("Please input a frame number greater 1!");
+                integerCheck = false;
+            }
+            return integerCheck;
+        }
+
+        public static bool CheckBracingInput(string input)
```
[the remark this answers is not in the document]
Comment (Rebecca-Coding21) [reply examples/Steel_Hall/Program.cs]: Implemented

File: examples/Steel_Hall/Program.cs [hunk 5]
```diff
@@ -0,0 +1,1291 @@
...
+            {
+                if (string.IsNullOrEmpty(input))
+                {
+                    throw new ArgumentNullException();
+                }
+                else if (!(input.ToLower() == "y" || input.ToLower() == "n"))
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                input.ToLower();
```
[the remark this answers is not in the document]
Comment (Rebecca-Coding21) [reply examples/Steel_Hall/Program.cs]: Implemented

File: examples/Steel_Hall/Program.cs [hunk 6]
```diff
@@ -0,0 +1,1291 @@
...
+            bool numberCheck = false;
+            string? inputValue = string.Empty;
+            string? inputNumber = string.Empty;
+            Bracing bracing = new Bracing();
+
+            while (check == false)
+            {
+                Console.Write(message);
+                inputValue = Console.ReadLine();
+                check = CheckBracingInput(inputValue);
```
[the remark this answers is not in the document]
Comment (Rebecca-Coding21) [reply examples/Steel_Hall/Program.cs]: Implemented

File: examples/Steel_Hall/Program.cs [hunk 7]
```diff
@@ -0,0 +1,1291 @@
...
+                int member_load_id = 1;
+
+                //member loads load case 2/5
+                foreach (var memberItem in xMembers)
+                {
+                    member_load newMemberLoad = new member_load()
+                    {
+                        no = member_load_id,
+                        members_string = memberItem.Key.ToString(),
+                        members = new int[1] { memberItem.Key },
```
[the remark this answers is not in the document]
Comment (Rebecca-Coding21) [reply examples/Steel_Hall/Program.cs]: Ichanged it that way.

File: examples/Continuous_Beam/Program.cs (hunk 3 above)
Comment (Rebecca-Coding21) [reply examples/Continuous_Beam/Program.cs:632]: I implemented that change into the first pull request

**Pull request closed without merging**